### DISCUSSION TITLE
Display location depth in prompt

### DISCUSSION THREAD

**Discussion by dpokluda** [JanDeDobbeleer/oh-my-posh#570] [Troubleshoot] [answered]
# Prerequisites
- [x] I have read and understand the CONTRIBUTING guide
- [x] I looked for duplicate issues before submitting this one

# Description
I have created my new theme but one thing that I can't figure out how to do, is that I would like to include the depth of my current location stack. For example in my existing (oh-my-posh v2) theme I used the following:

```
$GitPromptSettings.DefaultPromptSuffix = '$(''+'' * ((get-location -stack).count))'
```

Is there a way to achieve something like that? I have already looked at `command` block but for that I would need to start a new Powershell session which would reset my location stack.

**Comment by JanDeDobbeleer** [comment, marked as answer]
You can make use of the [environment](https://ohmyposh.dev/docs/environment) segment and use `Set-PoshContext` to add the stack count to an environment variable.

```pwsh
function Set-EnvVar {
    $env:STACK_COUNT = '$(''+'' * ((get-location -stack).count))'
}

New-Alias -Name 'Set-PoshContext' -Value 'Set-EnvVar' -Scope Global
```

**Comment by dpokluda** [comment]
Works perfectly, I wasn't aware of the fact that `Set-PoshContext` is available and called. This is very neat.

Thanks.
